atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraService.java

```diff
@@ -104,7 +105,8 @@ public CassandraService(
         this.blacklist = blacklist;
         this.poolMetrics = poolMetrics;
 
-        Supplier<Map<String, String>> hostnamesByIpSupplier = new HostnamesByIpSupplier(this::getRandomGoodHost);
+        Supplier<Map<String, String>> hostnamesByIpSupplier =
+                new HostnamesByIpSupplier(this::getAllHostsUnlessBlacklisted);
         this.hostnameByIpSupplier = Suppliers.memoizeWithExpiration(hostnamesByIpSupplier::get, 2, TimeUnit.MINUTES);
```

Comment:
How does this 2 minute memoization factor into cassandra nodes coming into and leaving cluster, especially during rolling restarts and upgrades?

Reply:
I don't have the historical context on why the 2 minutes was picked here, but from what I have seen, it takes roughly 2 (usually 3) minutes for a Cassandra node to leave/re-join a cluster, so this should be safe. Although, maybe we knock this down to 1 minute? Question for the Atlas folks!

Reply:
No strong opinions on the memoisation length - it was probably picked arbitrarily, honestly. Yeah we can make this 1.